### PR TITLE
feat(frontend): update datahub-web client UI code

### PR DESCRIPTION
**Pull request by cptran777**
This is the current state of open-sourcable UI code in datahub-web. Large scale migration of code, removal of internal only logic that pollutes craftsmanship of UI code, additional features and entities support. 

Some things that are included now are:

- More extensible UI models as well as better integration with the API
- Lots of internal code that was leaked and non-functional in the open source package has been removed to reduce confusion and make a cleaner experience
- Better usage of configuration to setup UI rendering logic
- More generic use of components and unifying the user experience throughout the application
- Better pipeline and methodology for more consistent rollout of changes to UI code

Note that the above are all work in progress, but that's the main reason for the massive file change

**Testing Done:**

`./gradlew datahub-web:emberWorkspaceTest` results:

```
Done in 601.91s.

BUILD SUCCESSFUL in 10m 13s
```

## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

**Comment by cobolbaby** [comment]
What specific content has been changed in web-ui? Why are so many files changed?

**Comment by mars-lan** [comment]
Have you tested this end-to-end using the docker images? @jplaisted I assume we don't have a dev image for frontend yet?

**Comment by cptran777** [comment]
> What specific content has been changed in web-ui? Why are so many files changed?

Hey @cobolbaby , effectively the UI code on github has been stale/behind for about 6+ months. In that time, internally we made a lot of changes for a friendlier open source experience. Lots of code was cleaned up and migrated. 

Some things that are included now are:

- More extensible UI models as well as better integration with the API
- Lots of internal code that was leaked and non-functional in the open source package has been removed to reduce confusion and make a cleaner experience
- Better usage of configuration to setup UI rendering logic
- More generic use of components and unifying the user experience throughout the application
- Better pipeline and methodology for more consistent rollout of changes to UI code

Note that the above are all work in progress, but that's the main reason for the massive file change

**Comment by cobolbaby** [comment]
Does this pull request involve the following features?

- #1730 
- #1728

**Comment by cptran777** [comment]
> Does this pull request involve the following features?
> 
> * #1730
> * #1728

I'll have to get back to you on * #1728 , as I'm not sure I understand the issue. Will follow up with @mars-lan to get more context. As for * #1730 unfortunately this PR only catches up to state on internal code changes, but that is a new feature we still have in our backlog.

File: datahub-web/@datahub/data-models/LICENSE.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 The MIT License (MIT)
```
Comment (mars-lan) [inline datahub-web/@datahub/data-models/LICENSE.md:1]: Is this our license? If so it should Apache instead

**Comment by jplaisted** [comment]
> Have you tested this end-to-end using the docker images? @jplaisted I assume we don't have a dev image for frontend yet?

Correct. I was not familiar enough with our frontend stack to make one in that initial PR; it will require a little more dedicated time from me and/or Charlie. Let me make an issue to track it.

**Comment by jplaisted** [comment]
2,600 files changed is... a lot. Are any of these auto generated?

**Comment by cptran777** [comment]
> 2,600 files changed is... a lot. Are any of these auto generated?

They're only auto generated in the sense that our script took this directly from the current internal state.

**Comment by jplaisted** [comment]
> > 2,600 files changed is... a lot. Are any of these auto generated?
> 
> They're only auto generated in the sense that our script took this directly from the current internal state.

Yeah I meant generated on every build.

You're saying we have 2,600 hand written files?

**Comment by jplaisted** [comment]
> datahub-web/@nacho-ui

Why is this here? Is this experimental code (nacho referring to @igbopie ?)? It shouldn't be checked in internally, much less externally...

Also I see a lot of JS files. Are we not fully on TS yet? If those JS files are generated from TS we can delete them, right?

**Comment by cptran777** [comment]
> > > 2,600 files changed is... a lot. Are any of these auto generated?
> > 
> > 
> > They're only auto generated in the sense that our script took this directly from the current internal state.
> 
> Yeah I meant generated on every build.
> 
> You're saying we have 2,600 hand written files?

So I can't give an exact ratio, but there's a large chunk of code that gets produced by script when initializing a new module (i.e. configuration files and such). They're not produced on every build, but they can be thought of as an MP skeleton framework of sorts. But other than that, yeah, this is what happens when the UI code isn't touched for 6 months haha.

**Comment by cptran777** [comment]
> > datahub-web/@nacho-ui
> 
> Why is this here? Is this experimental code (nacho referring to @igbopie ?)? It shouldn't be checked in internally, much less externally...
> 
> Also I see a lot of JS files. Are we not fully on TS yet? If those JS files are generated from TS we can delete them, right?

Because of the way the Ember build works under the hood, not all files can be in JS. (Ember pipeline works some voodoo magic and builds a working tree from TS and configuration type files from all the modules and the main host application and does a lot of merging to get you the final output. A lot of these configuration type files cannot be in TS or else they aren't recognized by the voodoo magic of Ember). 

However, all code that we ourselves should be modifying (i.e. not generated boiler plate) should be in TS. 

Also, @nacho-ui is actually a library for shared components across the internal data apps that we are currently housing in our own repo due to logistics reasons. The name is definitely a joke on our actual teammate, but the code in there is very real and un-experimental, and necessary for the application to actually run.

File: datahub-web/@datahub/data-models/LICENSE.md (hunk 1 above)
Comment (cptran777) [reply datahub-web/@datahub/data-models/LICENSE.md]: Haha, default ember I guess. Changed.

**Comment by cptran777** [comment]
With the latest commit, the UI build is currently in a working state, confirmed end to end using docker images.

File: datahub-web/@datahub/data-models/addon/constants/entity/person/tabs.ts [hunk 2]
```diff
@@ -9,13 +9,64 @@ export enum PersonTab {
...
+  UserLikeList = 'userlikelist',
+  UserSocialActionList = 'usersocialactionlist',
+  UserDYMIList = 'userdymilist'
 }
 
 /**
  * Tab properties available for Person entity
  */
-export const personTabProperties: Array<ITabProperties> = [TabProperties.userlists, TabProperties.userownership];
+export const personTabProperties: Array<ITabProperties> = [
```
Comment (igbopie) [inline datahub-web/@datahub/data-models/addon/constants/entity/person/tabs.ts:21]: Doesnt this mean, we are exposing UMP flows as well as JIT? can you double check?

Comment (cptran777) [reply datahub-web/@datahub/data-models/addon/constants/entity/person/tabs.ts:21]: We don't as the actual render props only take in UserOwnership. That being said, this is a minor leak and we'll have to figure out a better way to do tab properties in the future to address this.

**Comment by cptran777** [comment]
Will be merging this in if there are no further comments.

**Pull request merged**